### PR TITLE
SONARJAVA-1087 Remove usage of AstNode in IndentationCheck

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java [hunk 1]
```diff
@@ -66,7 +67,8 @@ public void visitClass(ClassTree tree) {
   private boolean isPublicStaticNotFinal(VariableTree tree) {
     boolean isPublic = false, isStatic = false, isFinal = false;
 
-    for (Modifier modifier : tree.modifiers().modifiers()) {
+    for (ModifierKeywordTree modifierKeywordTree : tree.modifiers().modifiers()) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java]: New BLOCKER Impossible cast from org.sonar.plugins.java.api.tree.Modifier to org.sonar.plugins.java.api.tree.ModifierKeywordTree in org.sonar.java.checks.PublicStaticFieldShouldBeFinalCheck.isPublicStaticNotFinal(VariableTree) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)

File: java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java [hunk 2]
```diff
@@ -99,7 +100,8 @@ private void checkName(VariableTree variableTree) {
   private boolean isStaticFinal(VariableTree variableTree) {
     boolean isStatic = false;
     boolean isFinal = false;
-    for (Modifier modifier : variableTree.modifiers().modifiers()) {
+    for (ModifierKeywordTree modifierKeywordTree : variableTree.modifiers().modifiers()) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java]: New BLOCKER Impossible cast from org.sonar.plugins.java.api.tree.Modifier to org.sonar.plugins.java.api.tree.ModifierKeywordTree in org.sonar.java.checks.BadConstantName_S00115_Check.isStaticFinal(VariableTree) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

File: java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java (hunk 1 above)
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java]: BLOCKER Impossible cast from org.sonar.plugins.java.api.tree.Modifier to org.sonar.plugins.java.api.tree.ModifierKeywordTree in org.sonar.java.checks.PublicStaticFieldShouldBeFinalCheck.isPublicStaticNotFinal(VariableTree) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)

File: java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java (hunk 2 above)
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java]: BLOCKER Impossible cast from org.sonar.plugins.java.api.tree.Modifier to org.sonar.plugins.java.api.tree.ModifierKeywordTree in org.sonar.java.checks.BadConstantName_S00115_Check.isStaticFinal(VariableTree) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

File: java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java (hunk 1 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java]: False positive. One of the previous commit change the return type of `tree.modifiers().modifiers()` to `ModifierKeywordTree`

File: java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java (hunk 2 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/BadConstantName_S00115_Check.java]: False positive. One of the previous commit change the return type of `tree.modifiers().modifiers()` to `ModifierKeywordTree`

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 blocker

**Comment by benzonico** [comment]
LGTM. That's a load of code but seems somehow unavoidable. Might worth to explore deeper (but later) the idea of firstToken/lastToken methods on Tree interface.

**Pull request merged**